From f9583e98c2a73f1642c28dc7c043bc8a656c6b83 Mon Sep 17 00:00:00 2001
From: Zbigniew Lukwinski <zbigniew.lukwinski@linux.intel.com>
Date: Wed, 11 May 2022 15:26:24 +0200
Subject: [PATCH] i3c: mctp: workaround for wrong DCR value

In GNR A0 there is an issue about DCR value. I3C MNG IP in GNR A0
introduces itself with DCR=0xC9. Adding this value to allow MCTP over
I3C driver to be loaded in this case as well.
Issue will be fixed in GNR B0.

Signed-off-by: Zbigniew Lukwinski <zbigniew.lukwinski@linux.intel.com>
---
 drivers/i3c/mctp/i3c-mctp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/i3c/mctp/i3c-mctp.c b/drivers/i3c/mctp/i3c-mctp.c
index b0e3814..42e864d 100644
--- a/drivers/i3c/mctp/i3c-mctp.c
+++ b/drivers/i3c/mctp/i3c-mctp.c
@@ -333,6 +333,8 @@ static void i3c_mctp_remove(struct i3c_device *i3cdev)
 
 static const struct i3c_device_id i3c_mctp_ids[] = {
 	I3C_CLASS(0xCC, 0x0),
+	/* Workaround for GNR A0 where I3C MNG IP introduces itself with DCR=0xC9 */
+	I3C_CLASS(0xC9, 0x0),
 	{ },
 };
 
-- 
2.7.4

